src/System.Management.Automation/engine/CommandInfo.cs

```diff
@@ -928,7 +928,7 @@ private PSSyntheticTypeName(string typeName, Type type, IList<PSMemberNameAndTyp
         : base(typeName, type)
         {
             Members = membersTypes;
-            if (type != typeof(PSObject))
+            if (type != typeof(PSCustomObject))
```

Comment:
Above in lines 914 and 916 I see two Create() methods with _7 uses_ - all of them are subject to this change. Please describe all of them in the PR description (not only Hash).